### ISSUE TITLE
Regression `Gen.nonEmptyBuildableOf` and dependent methods since 1.14.3

### ISSUE THREAD

**Issue by satorg** [typelevel/scalacheck#714]
Consider the following example:
```scala
val nestedGen = Gen.listOf(Gen.resize(5, Gen.nonEmptyListOf(Gen.asciiChar)))

Prop
  .forAll(nestedGen, nestedGen, nestedGen, nestedGen, nestedGen, nestedGen, nestedGen, nestedGen) {
    (_, _, _, _, _, _, _, _) => true
  }
  .check()
```
It usually succeeds on 1.14.3, but almost always fails on 1.15.1 with errors like:
> Gave up after only 43 passed tests. 501 tests were discarded.

My guess is that `nonEmptyBuildableOf` has changed and that made the new implementation less reliable in some cases with a few of non-empty-whatever generators with reduces max size.

See also the initial conversation [here](568#issuecomment-725221811).

**Comment by satorg** [comment]
Similar issue: #581.
But it was posted for 1.14.x and it seems things became worse since then.

**Comment by ashawley** [comment]
It appears Erik caught this and fixed it, see #709.  It didn't get merged in time to be included in 1.15.1.   It will need to be released in the next version, 1.15.2.

**Comment by SethTisue** [comment]
so we should close this, then?

**Comment by satorg** [comment]
I guess yes – looks like the fix in #709 should resolve this issue as well. Thanks for pointing it out.
Looking forward to get it released:)
